File: Problem2.js
/**
 * Created by sle7 on 10/1/2015.
 */

/*
    Even Fibonacci Numbers
 */

var max = 4000000;

var sumFibEven = function(num) {
  var numA = 1;
  var numB = 2;
  var numNext = 0;
  var sum = 2;

  while (numNext < max) {
    numNext = numA + numB;
    if (numNext % 2 === 0) {
      sum += numNext;
    }
    numA = numB;
    numB = numNext;
  }
  return sum;
};

console.log(sumFibEven(max));

File: Problem2.rb
# Created by sle7 on 12/1/2015.

#
#    Even Fibonacci Numbers
#

a = 1
b = 2
n = 0
sum = 0
max = 4000000

while b < max do
  if b % 2 == 0
    sum += b
  end
  n = a + b
  a = b
  b = n
end

puts sum
